File: topic_5/Python/13-quadperf.py
# Recebe um número inteiro positivo e retorna True se for um 
# quadrado perfeito e Falso caso contrário

def quadperf(n):
    raiz = int(n ** (1/2))
    if (raiz ** 2) == n:
        return True
    return False

print(quadperf(12) == False) #True
print(quadperf(16) == True) #True
print(quadperf(25) == True) #True
print(quadperf(5) == False) #True

File: topic_5/Haskell/13-quadperf.hs
-- Recebe um número inteiro positivo e retorna True se for um 
-- quadrado perfeito e Falso caso contrário

quadperfList n = [x | x <- [1..n], x * x == n] -- Se for quadrado perfeito a lista não vai ser nula

quadperf n = if not $ null $ quadperfList n then True else False

main = do
    a <- readLn :: IO Int
    print $ quadperf a